File: locksmith/src/main/java/dk/nodes/locksmith/core/utils/HashingUtils.java
package dk.nodes.locksmith.utils;

import java.nio.charset.Charset;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;

import dk.nodes.locksmith.BuildConfig;

public class HashingUtils {
    public static String doHash(String text, String CHAR_SET) {
        try {
            MessageDigest digest = MessageDigest.getInstance("SHA-256");

            byte[] result = digest.digest(text.getBytes(Charset.forName(CHAR_SET)));

            StringBuilder sb = new StringBuilder();

            for (byte b : result) {
                sb.append(String.format("%02X", b));
            }

            return sb.toString();
        } catch (NoSuchAlgorithmException e) {
            if (BuildConfig.DEBUG) {
                e.printStackTrace();
            }
            return text;
        }
    }
}

File: nencrypt/src/main/java/dk/nodes/nencrypt/utils/HashingUtils.kt
package dk.nodes.nencrypt.utils

import dk.nodes.nencrypt.BuildConfig
import java.io.UnsupportedEncodingException
import java.security.MessageDigest
import java.security.NoSuchAlgorithmException


class HashingUtils {
    companion object {
        fun doHash(text: String, CHAR_SET: String): String {
            try {
                val digest = MessageDigest.getInstance("SHA-256")

                val result = digest.digest(text.toByteArray(charset(CHAR_SET)))

                val sb = StringBuilder()

                for (b in result) {
                    sb.append(String.format("%02X", b))
                }

                return sb.toString()
            } catch (e: NoSuchAlgorithmException) {
                if (BuildConfig.DEBUG) {
                    e.printStackTrace()
                }
                return text
            } catch (e: UnsupportedEncodingException) {
                if (BuildConfig.DEBUG) {
                    e.printStackTrace()
                }
                return text
            }
        }
    }
}